frame/staking-rewards/src/lib.rs

```diff
@@ -126,30 +130,41 @@ pub mod pallet {
 			amount: T::Balance,
 			/// Duration of stake.
 			duration_preset: DurationSeconds,
-			/// Position Id of newly created stake.
-			position_id: T::PositionId,
+			/// FNFT Collection Id
+			fnft_collection_id: T::AssetId,
+			/// FNFT Instance Id
+			fnft_instance_id: T::FinancialNftInstanceId,
 			keep_alive: bool,
 		},
 		Claimed {
 			/// Owner of the stake.
 			owner: T::AccountId,
-			/// Position Id.
-			position_id: T::PositionId,
+			/// FNFT Collection Id
+			fnft_collection_id: T::AssetId,
+			/// FNFT Instance Id
+			fnft_instance_id: T::FinancialNftInstanceId,
 		},
 		StakeAmountExtended {
-			position_id: T::PositionId,
+			/// FNFT Collection Id
+			fnft_collection_id: T::AssetId,
+			/// FNFT Instance Id
+			fnft_instance_id: T::FinancialNftInstanceId,
 			/// Extended amount
 			amount: T::Balance,
 		},
 		Unstaked {
 			/// Owner of the stake.
 			owner: T::AccountId,
-			/// Position Id of newly created stake.
-			position_id: T::PositionId,
+			/// FNFT Collection Id
+			fnft_collection_id: T::AssetId,
```

Comment:
Wonder if it's a good idea to combine these fields into a struct?

Reply:
I thought about this myself, we had them as a tuple originally but decided to split them up. a separate struct could be good, something like `struct FnftId { collection_id, instance_id }`

But considering they're always kept separate in the fnft pallet (and the various traits as well), I'm inclined to keep them separate here as well for consistency

Reply:
The only advantage I'd enjoy from combining them is to make the positionId a little more generic and upgradeable in the future if we ever choose to change how we query stakes (hopefully that is never necessary).

Reply:
Both good points.  But I guess we don't need to optimize it for now, having two separate as fNFT does is ok.

Reply:
@PoisonPhang we define position id as a tuple in staking currently:

```rust
type PositionId = (T::AssetId, T::FinancialNftInstanceId);
```

So we could in the future look into making this a named struct